### PR TITLE
[artifactory][artifactory-ha] Fixed broken issue - when setting `waitForDatabase: false` container startup still waits for DB

### PR DESCRIPTION
**Pull request by chukka**
#### PR Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] CHANGELOG.md updated
- [x] Variables and other changes are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[artifactory]`)

**What this PR does / why we need it**:
In case of external DB, We don't need to wait for DB to come up . When setting "waitForDatabase: false" container startup still waits for DB - This issue is fixed by adding an ENV variable `SKIP_WAIT_FOR_EXTERNAL_DB` set to true 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by danielezer — APPROVED** [review]

File: stable/artifactory-ha/templates/artifactory-node-statefulset.yaml [hunk 1]
```diff
@@ -214,6 +214,10 @@ spec:
         resources:
 {{ toYaml .Values.artifactory.node.resources | indent 10 }}
         env:
+      {{- if and (not .Values.waitForDatabase) (not .Values.postgresql.enabled) }}
+        - name: SKIP_WAIT_FOR_EXTERNAL_DB
+            value: "true"
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-node-statefulset.yaml:219]: Indent

File: stable/artifactory-ha/templates/artifactory-node-statefulset.yaml [hunk 2]
```diff
@@ -335,6 +339,10 @@ spec:
                 {{ tpl . $ }}
               {{- end }}
         env:
+      {{- if and (not .Values.waitForDatabase) (not .Values.postgresql.enabled) }}
+        - name: SKIP_WAIT_FOR_EXTERNAL_DB
+            value: "true"
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-node-statefulset.yaml:344]: Indent

File: stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml [hunk 3]
```diff
@@ -290,6 +290,10 @@ spec:
           mkdir -p {{ .Values.artifactory.persistence.mountPath }}/log;
           bash $scriptsPath/migrationStatus.sh {{ default .Chart.AppVersion .Values.artifactory.image.tag }} {{ .Values.artifactory.migration.timeoutSeconds }} > >(tee {{ .Values.artifactory.persistence.mountPath }}/log/helm-migration.log) 2>&1;
         env:
+      {{- if and (not .Values.waitForDatabase)  (not .Values.postgresql.enabled) }}
+        - name: SKIP_WAIT_FOR_EXTERNAL_DB
+            value: "true"
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml:295]: Indent

File: stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml [hunk 4]
```diff
@@ -429,6 +433,10 @@ spec:
                 {{ tpl . $ }}
               {{- end }}
         env:
+      {{- if and (not .Values.waitForDatabase) (not .Values.postgresql.enabled) }}
+        - name: SKIP_WAIT_FOR_EXTERNAL_DB
+            value: "true"
```
Comment (eldada) [inline stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml:438]: Indent

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 5]
```diff
@@ -239,6 +238,10 @@ spec:
           mkdir -p {{ .Values.artifactory.persistence.mountPath }}/log;
           bash $scriptsPath/migrationStatus.sh {{ default .Chart.AppVersion .Values.artifactory.image.tag }} {{ .Values.artifactory.migration.timeoutSeconds }} > >(tee {{ .Values.artifactory.persistence.mountPath }}/log/helm-migration.log) 2>&1;
         env:
+      {{- if and (not .Values.waitForDatabase) (not .Values.postgresql.enabled) }}
+        - name: SKIP_WAIT_FOR_EXTERNAL_DB
+            value: "true"
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-statefulset.yaml:243]: Indent

File: stable/artifactory/templates/artifactory-statefulset.yaml [hunk 6]
```diff
@@ -353,6 +356,10 @@ spec:
                 {{ tpl . $ }}
               {{- end }}
         env:
+      {{- if and (not .Values.waitForDatabase) (not .Values.postgresql.enabled) }}
+        - name: SKIP_WAIT_FOR_EXTERNAL_DB
+            value: "true"
```
Comment (eldada) [inline stable/artifactory/templates/artifactory-statefulset.yaml:361]: Indent

**Review by eldada — CHANGES_REQUESTED** [review]

File: stable/artifactory-ha/templates/artifactory-node-statefulset.yaml (hunk 2 above)
Comment (chukka) [reply stable/artifactory-ha/templates/artifactory-node-statefulset.yaml]: Fixed

File: stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml (hunk 3 above)
Comment (chukka) [reply stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml]: Fixed

File: stable/artifactory/templates/artifactory-statefulset.yaml (hunk 5 above)
Comment (chukka) [reply stable/artifactory/templates/artifactory-statefulset.yaml]: Fixed

**Review by eldada — APPROVED** [review]

File: stable/artifactory/templates/artifactory-statefulset.yaml (hunk 6 above)
Comment (chukka) [reply stable/artifactory/templates/artifactory-statefulset.yaml]: Fixed

**Review by rahulsadanandan — APPROVED** [review]

**Pull request merged**